python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -426,98 +434,74 @@ def __init__(
+                jobs, "jobs", (JobDefinition, UnresolvedAssetJobDefinition)
+            ),
+            resources=check.opt_mapping_param(resources, "resources", key_type=str),
+            executor=check.opt_inst_param(executor, "executor", (ExecutorDefinition, Executor)),
+            loggers=check.opt_mapping_param(
+                loggers, "loggers", key_type=str, value_type=LoggerDefinition
+            ),
+            asset_checks=check.opt_iterable_param(
+                asset_checks,
+                "asset_checks",
+                (AssetChecksDefinition, AssetsDefinition),
+            ),
         )
+        self._original_args = {
+            "assets": assets,
+            "schedules": schedules,
+            "sensors": sensors,
+            "jobs": jobs,
+            "resources": resources,
+            "executor": executor,
+            "loggers": loggers,
+            "asset_checks": asset_checks,
+        }
 
     @property
-    def assets(self) -> Iterable[Union[AssetsDefinition, SourceAsset, CacheableAssetsDefinition]]:
-        return self._assets
-
-    @property
-    def schedules(
-        self,
-    ) -> Iterable[Union[ScheduleDefinition, UnresolvedPartitionedAssetScheduleDefinition]]:
-        return self._schedules
-
-    @property
-    def sensors(self) -> Iterable[SensorDefinition]:
-        return self._sensors
-
-    @property
-    def jobs(self) -> Iterable[Union[JobDefinition, UnresolvedAssetJobDefinition]]:
-        return self._jobs
-
-    @property
-    def resources(self) -> Mapping[str, Any]:
-        return self._resources
-
-    @property
-    def executor(self) -> Optional[Union[ExecutorDefinition, Executor]]:
-        return self._executor
-
-    @property
-    def loggers(self) -> Mapping[str, LoggerDefinition]:
-        return self._loggers
-
-    @property
-    def asset_checks(self) -> Iterable[AssetChecksDefinition]:
-        return self._asset_checks
+    def original_args(self) -> DefinitionsArgs:
+        return self._original_args
 
     @public
     def get_job_def(self, name: str) -> JobDefinition:
         """Get a job definition by name. If you passed in a an :py:class:`UnresolvedAssetJobDefinition`
         (return value of :py:func:`define_asset_job`) it will be resolved to a :py:class:`JobDefinition` when returned
-        from this function.
+        from this function, with resources applied.
         """
         check.str_param(name, "name")
         return self.get_repository_def().get_job(name)
 
     @public
     def get_sensor_def(self, name: str) -> SensorDefinition:
-        """Get a sensor definition by name."""
+        """Get a :py:class:`SensorDefinition` by name.
+        If your passed-in sensor had resource dependencies, or the job targeted by the sensor had
+        resource dependencies, those will be resolved on the returned object. This object will not match the
+        :py:class:`SensorDefinition` passed into the constructor by reference.
+        """
```

Comment:
Seems like we could have a more concise docblock

Something like

"Return the sensor definition with all resources dependencies fully resolved"

Reply:
Fine with that.